backend/library/views.py

```
@@ -189,54 +219,91 @@ def upload_library(self, request):
             )
 
+    @action(detail=False, name="Get provider choices")
+    def provider(self, request):
+        providers = set(StoredLibrary.objects.all().values_list("provider", flat=True))
+        return Response({p: p for p in providers})
 
-class LoadedLibraryViewSet(viewsets.ModelViewSet):
+    @action(detail=False, name="Get locale choices")
+    def locale(self, request):
+        locales = set(
+            chain.from_iterable([l.get_locales for l in StoredLibrary.objects.all()])
```

Ruff (0.8.2)

Ambiguous variable name: `l`

(E741)